File: apps/admin-web/src/api/article.ts
import client from "./client";

export interface ArticleDraft {
  id?: number;
  title: string;
  slug: string;
  description: string;
  content: string;
  cover: string;
  category: string;
  tags: string[];
  draft: boolean;
  pinned: boolean;
  comment_enabled: boolean;
  toc_enabled: boolean;
  word_count?: number;
  published_at?: string;
  updated_at?: string;
  created_at?: string;
}

export function getArticles(params?: Record<string, any>) {
  return client.get("/articles", { params });
}

export function getArticle(id: number) {
  return client.get(`/articles/${id}`);
}

export function createArticle(data: ArticleDraft) {
  return client.post("/articles", data);
}

export function updateArticle(id: number, data: ArticleDraft) {
  return client.put(`/articles/${id}`, data);
}

export function deleteArticle(id: number) {
  return client.delete(`/articles/${id}`);
}

export function publishArticle(id: number) {
  return client.post(`/articles/${id}/publish`);
}

export function unpublishArticle(id: number) {
  return client.post(`/articles/${id}/unpublish`);
}

File: apps/admin-api/models/article.go
package models

import "time"

type ArticleDraft struct {
	ID              uint       `gorm:"primaryKey" json:"id"`
	Title           string     `gorm:"not null" json:"title"`
	Slug            string     `gorm:"not null" json:"slug"`
	Description     string     `json:"description"`
	Content         string     `gorm:"not null" json:"content"`
	Cover           string     `json:"cover"`
	Category        string     `json:"category"`
	TagsJSON        string     `json:"tags_json"` // JSON array stored as string
	Draft           bool       `gorm:"default:true" json:"draft"`
	Pinned          bool       `gorm:"default:false" json:"pinned"`
	CommentEnabled  bool       `gorm:"default:true" json:"comment_enabled"`
	TocEnabled      bool       `gorm:"default:true" json:"toc_enabled"`
	ContentVersion     int        `gorm:"default:1" json:"content_version"`
	WordCount          int        `gorm:"default:0" json:"word_count"`
	AiSummary          string     `json:"ai_summary"`
	AiSummaryGeneratedAt *time.Time `json:"ai_summary_generated_at"`
	AiSummaryModel     string     `json:"ai_summary_model"`
	PublishedAt        *time.Time `json:"published_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
	CreatedAt       time.Time  `json:"created_at"`
}

func (ArticleDraft) TableName() string {
	return "article_draft"
}

type ArticlePublishRecord struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	DraftID       uint      `json:"draft_id"`
	Title         string    `gorm:"not null" json:"title"`
	Slug          string    `gorm:"not null" json:"slug"`
	MarkdownPath  string    `gorm:"not null" json:"markdown_path"`
	PublishedAt   time.Time `json:"published_at"`
	BuildRecordID *uint     `json:"build_record_id"`
}

func (ArticlePublishRecord) TableName() string {
	return "article_publish_record"
}
